src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexReadOperation.cs

```diff
@@ -51,12 +48,17 @@ public CoraxIndexReadOperation(Index index, Logger logger, Transaction readTrans
             if (isDistinctCount)
                 pageSize = int.MaxValue;
             var position = query.Start;
-
-            IQueryMatch result = _coraxQueryEvaluator.Search(query, fieldsToFetch);
+            var take = Math.Min(pageSize, Convert.ToInt32(_indexSearcher.CountItemsInIndex)) + position;
```

Comment:
1. Shouldn't we modify `GetPageSize()` method of the base class (to pass number of entries of the searcher instead of the searcher instance) and use it here?

https://github.com/ravendb/ravendb/blob/d9f052d6229062b7ee2ec9fd2d2f193c36943241/src/Raven.Server/Documents/Indexes/Persistence/IndexOperationBase.cs#L261-L270

Reply:
2. Is it correct that we add `position` which is `query.Start` to `take` parameter?

Reply:
2. Take is a special parameter for query optimization inside Corax. It's a reference to how many values we want to cache while querying.  For example, when doing ordering we need to cache **query.Start+position** to persist previous result (query.Start) + how much we want to extend our array.

1. Good catch, I've changed but to something else. If  **query.Start+position  > CountItemsInIndex** just set take to -1 (const from Corax)